src/app/beer_garden/app.py

```diff
@@ -152,6 +154,27 @@ def handle_event(event):
                 and event.metadata["entry_point_type"] == "HTTP"
             ):
                 beer_garden.local_plugins.manager.rescan()
+        if (event.name == "COMMAND_PUBLISHING_BLOCKLIST_SYNC") and (
+            event.garden != config.get("garden.name")
+        ):
+            local_command_publishing_blocklist = CommandPublishingBlocklist.objects(
```

Comment:
All of the logic in this if block should really be in a function that lives elsewhere, presumably in `beer_garden.command_publishing_blocklist`.  Similar to how the code above this just called `beer_garden.local_plugins.manager.rescan()`, rather than having all of the rescan logic mashed in here.

Reply:
Also though, I don't think this is the right place for this handling to live.  It should be in [`beer_garden.events.handlers`](https://github.com/beer-garden/beer-garden/blob/3.13.0/src/app/beer_garden/events/handlers.py#L37).  In there you would tack on an entry like:

```python
(beer_garden.command_publishing_blocklist.handle_event, "CommandPublishingBlocklist event handler"),
```

And that `handle_event` method would essentially contain your `event.name` and `event.garden` check, followed by this logic.